libvcx/src/v3/handlers/connection/connection.rs

```diff
@@ -102,22 +103,25 @@ impl Connection {
 
         let messages = self.get_messages()?;
         trace!("Connection::update_state >>> retrieved messages {:?}", messages);
-        let agent_info = self.agent_info().clone();
 
         if let Some((uid, message)) = self.connection_sm.find_message_to_handle(messages) {
+            trace!("Connection::update_state >>> handling message uid: {:?}", uid);
             self.handle_message(message.into())?;
-            agent_info.update_message_status(uid)?;
-        };
-
-        if let Some(prev_agent_info) = self.connection_sm.prev_agent_info().cloned() {
+            self.agent_info().clone().update_message_status(uid)?;
+        } else if let Some(prev_agent_info) = self.connection_sm.prev_agent_info().cloned() {
```

Comment:
Previously this `prev_agent_info` could execute even if previous executed, now can't. Why is it safe to change this logic? I have suspect this might had to do with redirection feature?

Reply:
Resolved via DMs.